AdyenNetworking/APIClient/APIClient.swift

```diff
@@ -20,7 +20,7 @@ public protocol APIClientProtocol: AnyObject {
     
     /// :nodoc:
     /// Performs the API download request.
-    func perform<R: Request>(downloadRequest request: R, completionHandler: @escaping CompletionHandler<R.ResponseType>) where R.ResponseType == DownloadResponse
+    func perform<R: Request>(_ request: R, completionHandler: @escaping CompletionHandler<R.ResponseType>) where R.ResponseType == DownloadResponse
```

Comment:
I kind of like the explicitness of it, if you have a download request, which version will be called?

because a download request matches both the classic perform function and the new download one, so the caller has to clarify the ambiguity somehow, right?

I would say be explicit about it to prevent confusion and mistakes

Reply:
Yeah so you're right in that it's less explicit on the face of it, but it actually allows for easier integration when there is a single caller function. Swift is quite clever in that it will resolve to the most strict match. Meaning if the inferred type is of DownloadResponse, it will pick the correct overloaded implementation for downloading.

Reply:
Agree here with @mcguire-james 

It makes AplClient's API agnostic to the type of request you are making, so as a consumer you don't need to remember what type of ResponseType you Request have.

Reply:
By the way, XCode is not very smart.

![ezgif-3-df2a3212fe](https://user-images.githubusercontent.com/2648655/193688546-4739b919-169b-4be3-a8b6-1e0127350ce8.gif)

Reply:
<img width="1050" alt="Screen Shot 2022-10-04 at 11 41 45" src="https://user-images.githubusercontent.com/2619933/193787891-d67e7960-1ca4-458e-b32f-23b12d008100.png">

so this is what I don't like, by looking at the autocompletion list, I have no idea which which one is the download one :-)

Reply:
<img width="768" alt="Screen Shot 2022-10-04 at 12 40 44" src="https://user-images.githubusercontent.com/17139737/193799329-f8c92a3f-cad5-43ec-9ceb-dcfec51f05d2.png">
Weird because the async one shows it.

Reply:
oh, I am still using Xcode 13.4, cool, I will upgrade to Xcode 14 then :-D, ok, then it should be fine :-)